src/mbgl/renderer/layers/render_line_layer.cpp

```diff
@@ -242,7 +242,11 @@ GeometryCollection offsetLine(const GeometryCollection& rings, double offset) {
             Point<double> extrude = util::unit(aToB + bToC);
 
             const double cosHalfAngle = extrude.x * bToC.x + extrude.y * bToC.y;
-            extrude *= (1.0 / cosHalfAngle);
+            if (cosHalfAngle == 0) {
```

Comment:
if cosHalfAngle is anyway never supposed to be 0, would 
if (cosHalfAngle > 0)
  extrude *= (1.0 / cosHalfAngle);
be enough to avoid ub?

Reply:
when `i == ring.end() -1; bToC = zero; cosHalfAngle = 0;` 
so it is possible to have `cosHalfAngle == 0;`

Reply:
`extrude = cosHalfAngle ? (1.0 / cosHalfAngle) : zero;`